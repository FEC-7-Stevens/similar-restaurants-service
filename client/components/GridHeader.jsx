import React from 'react';
import styled from 'styled-components';

const Title = styled.h3`
  background-color: green;
  width: 1408px;
  height: 32px;
  margin-bottom: 25px;
  padding-top: 6px;
  border-top: 2px solid black;
  `;

const GridHeader = ({ currentCategory, currentName }) => (
  <div>
    <Title>
      {'More '}
      {currentCategory}
      {' Near '}
      {currentName}
    </Title>
  </div>
);

export default GridHeader;
