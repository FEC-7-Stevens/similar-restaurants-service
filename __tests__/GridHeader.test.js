import React from 'react';
import renderer from 'react-test-renderer';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import GridHeader from '../client/components/GridHeader';

configure({ adapter: new Adapter() });

describe('GridHeader', () => {
  const gridHeaderProps = {
    currentCategory: 'Pizza',
    currentName: 'Rubie ivory consequatur',
  };

  it('renders without crashing', () => {
    shallow(<GridHeader {...gridHeaderProps} />);
  });

  it('renders text correctly', () => {
    const title = shallow(<GridHeader {...gridHeaderProps} />);
    expect(title.find('div').text()).toEqual('More Pizza Near Rubie ivory consequatur');
  });

  it('GridHeader renders "More Pizza Near Rubie ivory consequatur"', () => {
    const component = renderer.create(<GridHeader {...gridHeaderProps} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
