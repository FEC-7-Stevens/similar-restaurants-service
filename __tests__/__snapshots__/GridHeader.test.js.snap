// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridHeader GridHeader renders "More Pizza Near Rubie ivory consequatur" 1`] = `
<div
  className="GridHeader__Title-sc-107enjf-0 kiVOvW"
>
  More 
  Pizza
   Near 
  Rubie ivory consequatur
</div>
`;
